src/analyses/libraryFunctions.ml

```diff
@@ -30,18 +35,37 @@ let posix_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
 let pthread_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     ("pthread_create", special [__ "thread" [w]; drop "attr" [r]; __ "start_routine" [s]; __ "arg" []] @@ fun thread start_routine arg -> ThreadCreate { thread; start_routine; arg }); (* For precision purposes arg is not considered accessed here. Instead all accesses (if any) come from actually analyzing start_routine. *)
     ("pthread_exit", special [__ "retval" []] @@ fun retval -> ThreadExit { ret_val = retval }); (* Doesn't dereference the void* itself, but just passes to pthread_join. *)
+    ("pthread_cond_wait", special [__ "cond" []; __ "lock" []] @@ fun cond lock -> CondWait { cond; lock });
+    ("pthread_cond_timedwait", special [__ "cond" []; __ "lock" []; drop "abstime" [r]] @@ fun cond lock -> CondWait { cond; lock });
   ]
 
 (** GCC builtin functions.
     These are not builtin versions of functions from other lists. *)
 let gcc_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
     ("__builtin_object_size", unknown [drop "ptr" [r]; drop' []]);
+    ("__builtin_prefetch", unknown (drop "addr" [] :: VarArgs (drop' [])));
+    ("__builtin_expect", special [__ "exp" []; drop' []] @@ fun exp -> Identity exp);
+    ("__builtin_unreachable", special' [] @@ fun () -> if get_bool "sem.builtin_unreachable.dead_code" then Abort else Unknown); (* https://github.com/sosy-lab/sv-benchmarks/issues/1296 *)
+    ("__assert_rtn", special [drop "func" [r]; drop "file" [r]; drop "line" []; drop "exp" [r]] @@ Abort); (* gcc's built-in assert *)
+    ("__builtin_return_address", unknown [drop "level" []]);
   ]
 
-(** Linux kernel functions. *)
-let linux_descs_list: (string * LibraryDesc.t) list = (* LibraryDsl. *) [
+let big_kernel_lock = AddrOf (Cil.var (Goblintutil.create_var (makeGlobalVar "[big kernel lock]" intType)))
+let console_sem = AddrOf (Cil.var (Goblintutil.create_var (makeGlobalVar "[console semaphore]" intType)))
 
-]
+(** Linux kernel functions. *)
+(* TODO: conditional on kernel option *)
+let linux_descs_list: (string * LibraryDesc.t) list = LibraryDsl.[
+    ("spin_lock_irqsave", special [__ "lock" []; drop "flags" []] @@ fun lock -> Lock { lock; try_ = get_bool "sem.lock.fail"; write = true; return_on_success = true });
+    ("spin_unlock_irqrestore", special [__ "lock" []; drop "flags" []] @@ fun lock -> Unlock lock);
+    ("_raw_spin_unlock_irqrestore", special [__ "lock" []; drop "flags" []] @@ fun lock -> Unlock lock);
+    ("spinlock_check", special [__ "lock" []] @@ fun lock -> Identity lock);
```

Comment:
Maybe a comment why this is Identity would be helpful here?

Reply:
Added!